### PR TITLE
feat(Password): move `InputPassword` to `Field.Password`

### PR DESCRIPTION
**Pull request by joakbjerk**
closes https://github.com/dnbexperience/eufemia/issues/3269

- [x] Provide an "export" for backwards compatibility.
- [x] Add deprecated comment to the export.
- [x] Add v11 info about how to move over.
- [x] Rewrite to typescript and hooks.
- [x] Update types.
- [x] Integrate new ´Forms´-hooks.
- [x] Add documentation.
- [x] Fix error styling based on `hasError`
- [x] Remove non-functioning `useDataValue` hook

### CODE REVIEW

File: packages/dnb-eufemia/src/extensions/forms/Field/Password/Password.tsx [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+
+function Password(props: PasswordProps) {
+  const [hidden, setHidden] = useState<boolean>(true)
+
+  const sharedContext = useContext(SharedContext)
+  const translations = sharedContext.getTranslation(props).Forms
+
+  const errorMessages = useErrorMessage(props.path, props.errorMessages, {
+    required: translations.passwordRequired,
+    pattern: translations.passwordErrorPattern,
```
Comment (tujoworker) [inline packages/dnb-eufemia/src/extensions/forms/Field/Password/Password.tsx:73]:
```suggestion
    required: translations.inputErrorRequired,
    pattern: translations.inputErrorPattern,
```

Both are so edge cases, that I think we rather should use the generalized once.

**Review by tujoworker — COMMENTED** [review]
Nice work! I left a couple of comments for now.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-fields/Password/Examples.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          title: 'Help is available',
+          content:
+            'Helping others, encouraging others, are often acts of being kind that have more meaning that you may realize.',
+        }}
+      />
+    </ComponentBox>
+  )
+}
+
+export const Disabled = () => {
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-fields/Password/Examples.tsx:75]: Does not work/look as expected. But I don't think we need such an example either.

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-fields/Password/Examples.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
+import ComponentBox from '../../../../../../shared/tags/ComponentBox'
+import { Field, FormError } from '@dnb/eufemia/src/extensions/forms'
+
+export const Empty = () => {
```
Comment (tujoworker) [inline packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-fields/Password/Examples.tsx:4]: I don't think we need this example. Label should always be used.

**Comment by joakbjerk** [comment]
> Nice work! I left a couple of comments for now.

Thank you 😁 will take a proper look tomorrow. Stuck in inovation day, for the rest of today 🙏

File: packages/dnb-eufemia/src/extensions/forms/Field/String/String.tsx [hunk 4]
```diff
@@ -183,6 +185,10 @@ function StringComponent(props: Props) {
     value: transformInstantly(value?.toString() ?? ''),
   }
 
+  if (props['aria-describedby']) {
```
[the remark this answers is not in the document]
Comment (joakbjerk) [reply packages/dnb-eufemia/src/extensions/forms/Field/String/String.tsx:188]: Yeah I agree, smoother solution than this re-assignment "hack" 🔥

File: packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-fields/Password/Examples.tsx (hunk 2 above)
Comment (joakbjerk) [reply packages/dnb-design-system-portal/src/docs/uilib/extensions/forms/feature-fields/Password/Examples.tsx:61]: I have fixed the disabled functionality now, so it should work as intended. Maybe you could take another look and see if the example is still needed or not?

**Review by tujoworker — APPROVED** [review]
Nice work! Thank you! ⭐

**Pull request merged**